ros2_easy_test/env.py

```diff
@@ -53,17 +53,26 @@ class ROS2TestEnvironment(Node):
     Args:
         watch_topics: The topics (and their type) to watch for incoming messages.
             You may pass ``None`` to not watch any topic.
+        qos_profile: The topics and their qos profile to subscribe to or publish on.
+            You may pass ``None`` to use the default profile.
     """
 
-    def __init__(self, *, watch_topics: Optional[Mapping[str, Type]] = None, **kwargs) -> None:
+    def __init__(
+        self,
+        *,
+        watch_topics: Optional[Mapping[str, Type]] = None,
+        qos_profiles: Optional[Mapping[str, QoSProfile]] = None,
+        **kwargs,
+    ) -> None:
         super().__init__("_testing_ROS2TestEnvironment", **kwargs)
 
         # Collects the messages that were received
         # (Simple)Queues are synchronized, but we need the lock to synchronize access to the mailbox dict
         self._subscriber_mailboxes_lock = RLock()
         self._subscriber_mailboxes: Dict[str, SimpleQueue[RosMessage]] = {}
 
-        self._qos_profile = QoSProfile(history=QoSHistoryPolicy.KEEP_ALL)
+        self._qos_profiles = qos_profiles or {}
+        self._default_qos_profile = QoSProfile(history=QoSHistoryPolicy.KEEP_ALL)
```

Comment:
This is probably a good use of a `collections.defaultdict`.

Reply:
I initially didn't want to use a defaultdict because it creates the same entry with the default profile for each topic. But that's probably premature optimization and a defaultdict is less error prone. I changed it to use a defaultdict